source/agora/consensus/data/Transaction.d

```diff
@@ -268,3 +294,46 @@ public bool getSumOutput (const Transaction tx, ref Amount acc)
             return false;
     return true;
 }
+
+unittest
+{
+    import vibe.data.json;
+    Transaction old_tx = Transaction(
+        TxType.Data,
+        [Input(Hash.init, 0)],
+        [Output.init],
+        [1,2,3]
+    );
+    auto json_str = old_tx.serializeToJsonString();
+
+    Transaction new_tx = deserializeJson!Transaction(json_str);
+    assert(new_tx.data.length == old_tx.data.length);
+    assert(new_tx.data == old_tx.data);
+}
+
+/*******************************************************************************
+
+    Caculates the fee of transaction data to store
+
+    Params:
+        data_size = The size of the data
+        tx_fee = The fee of transaction
+        factor = The factor to calculate for the fee of transaction data
+
+    Return:
+        Fees payable to data storage
+
+*******************************************************************************/
+
+public Amount clculateDataFee(ulong data_size, ulong tx_fee, uint factor) @safe nothrow
+{
+    const double decimal = 100.0;
+    ulong fee = cast(ulong)((round((exp(cast(double)data_size / cast(double)factor) - 1.0) * decimal) / decimal) * 10_000_000L);
+    return Amount(fee + tx_fee);
+}
```

Comment:
The formula is as follows:
<img width="237" alt="" src="https://user-images.githubusercontent.com/11639939/97793908-fee1fe00-1c35-11eb-8610-4002480eacb2.png">
<img width="644" alt="" src="https://user-images.githubusercontent.com/11639939/97793915-0ef9dd80-1c36-11eb-8ee7-89a501f683ff.png">

<img width="249" alt="" src="https://user-images.githubusercontent.com/11639939/97793919-19b47280-1c36-11eb-85f9-18715e664cf2.png">

Reply:
If we adjust the factor, we can change the fee appropriately.
I added factor to ConsensusParams.